src/Stratis.Bitcoin.Features.BlockStore/BlockStoreSignaled.cs

```diff
@@ -128,90 +142,104 @@ public void RelayWorker()
         {
             this.logger.LogTrace("()");
 
-            this.asyncLoop = this.asyncLoopFactory.Run($"{this.name}.RelayWorker", async token =>
+            this.relayWorkerTask = Task.Factory.StartNew(async delegate
             {
-                this.logger.LogTrace("()");
-
-                int announceBlockCount = this.blocksToAnnounce.Count;
-                if (announceBlockCount == 0)
+                while (!this.nodeLifetime.ApplicationStopping.IsCancellationRequested)
                 {
-                    this.logger.LogTrace("(-)[NO_BLOCKS]");
-                    return;
-                }
+                    // Wait until a new block is added to the queue.
+                    this.blockEnqueued.Wait();
 
-                this.logger.LogTrace("There are {0} blocks in the announce queue.", announceBlockCount);
+                    // Make sure that at least 'FlushFrequencySeconds' seconds passed since the last announcement.
+                    // This is needed in order to ensure announcing blocks in batches to reduce the overhead.
+                    double secondsSinceLastAnnounce = (DateTime.Now - this.lastBlockAnnounceTimeStamp).TotalSeconds;
+                    if (secondsSinceLastAnnounce < FlushFrequencySeconds)
+                        await Task.Delay(TimeSpan.FromSeconds(FlushFrequencySeconds - secondsSinceLastAnnounce));
+                    this.lastBlockAnnounceTimeStamp = DateTime.Now;
 
-                // Initialize this list with default size of 'announceBlockCount + 4' to prevent it from autoresizing during adding new items.
-                // This +4 extra size is in case new items will be added to the queue during the loop.
-                var broadcastItems = new List<ChainedBlock>(announceBlockCount + 4);
+                    this.blockEnqueued.Reset();
 
-                while (this.blocksToAnnounce.TryPeek(out ChainedBlock block))
-                {
-                    this.logger.LogTrace("Checking if block '{0}' is on disk.", block);
+                    this.logger.LogTrace("()");
```

Comment:
this is an entry log, it is supposed to be on method entry
same for lambdas and delagates - at their very start, not in the middle of nowhere

Reply:
Fixed.